plugins/chat/assets/javascripts/discourse/lib/chat-preload-link.js

```diff
@@ -0,0 +1,16 @@
+export function addPreloadLink(url) {
+  if (document.querySelector(`link[href="${url}"][rel="preload"]`)) {
+    return;
+  }
+
+  const importNode = document.createElement("link");
+  importNode.rel = "preload";
+  importNode.crossOrigin = "anonymous";
+  importNode.as = "fetch";
+  importNode.href = url;
+  importNode.onload = () => {
+    importNode?.classList?.add("is-prefetched");
+  };
+
+  document.head.appendChild(importNode);
```

Comment:
This is interesting, does adding a `<link rel=preload>` when the browser has already finished rendering the page do anything? My understanding of `<link rel=preload>` is that it is only useful for a full page load to prioritise resources to be loaded before the rendering engine kicks in.

Reply:
Yes I wasn't sure too, but it seems to work well yes

Reply:
Im curious to see if it works that well in production 🤞